### PR TITLE
f/threads: Update to file version 2.5

### PR DESCRIPTION
**Pull request by rockstorm101**
Obtained from:
https://dkprojects.net/openscad-threads/threads.scad

Fixes openscad/MCAD#79

### CODE REVIEW

**Comment by hyperair** [comment]
I think the two files have diverged significantly, so we shouldn't just blindly check-in the new version from openscad-threads.

File: fasteners/threads.scad [hunk 1]
```diff
@@ -1,245 +1,407 @@
...
+// taper       - diameter change per length (National Pipe Thread/ANSI B1.20.1
+//               is 1" diameter per 16" length). Taper decreases from 'diameter'
+//               as z increases.  Default: 0 (no taper).
+// leadin      - 0 (default): no chamfer; 1: chamfer (45 degree) at max-z end;
+//               2: chamfer at both ends, 3: chamfer at z=0 end.
+// leadfac     - scale of leadin chamfer length (default: 1.0 = 1/2 thread).
+// test        - true = do not render threads (just draw "blank" cylinder).
+//               Default: false (draw threads).
+module metric_thread (diameter=8, pitch=1, length=1, internal=false, n_starts=1,
+                      thread_size=-1, groove=false, square=false, rectangle=0,
```
Comment (hyperair) [inline fasteners/threads.scad:101]: I disagree with putting `square` and `rectangle` as parameters to `metric_thread` -- there are no square or rectangle metric threads, afaik.

**Comment by hyperair** [comment]
I'm closing this PR for now. Please open another if you wish to port the new features over from `openscad-threads`.

File: fasteners/threads.scad (hunk 1 above)
Comment (MichaelAtOz) [reply fasteners/threads.scad:101]: https://en.wikipedia.org/wiki/Square_thread_form

"Square threads are defined as follows by ISO standards:

    Sq 60 × 9 

where Sq designates a square thread, 60 is the nominal diameter in millimeters, and 9 is the pitch in millimeters. When there is no suffix it is a single start thread. If there is a suffix then the value after the multiplication sign is the lead and the value in the parentheses is the pitch. For example:

    Sq 60 × 18 ( P9 ) LH 

would denoted two starts, as the lead divided by the pitch is two. The "LH" denotes a left hand thread."

& pages 3-4 of [this](https://nptel.ac.in/content/storage2/courses/112105125/pdf/module-6%20lesson-1.pdf).

ISO standard and specified in millimeters, I'd say it is metric.

I'm not sure re rectangular, but I think it is nomenclature where the depth and pitch are not symmetrical.

Comment (hyperair) [reply fasteners/threads.scad:101]: Hmm, I stand corrected. I still don't think they should be part of the same module though.

**Comment by rockstorm101** [comment]
Hi @hyperair, thank you for reviewing this. I agree with your comments, particularly:
> I still don't think they should be part of the same module though.

and

> we use sweep() of a 2D object for a much more efficient implementation of threads [...] We also obey $fn, $fs, and $fa, which the other version does not.

So I'm happy to close this. I'll try and work out (hopefully with Dan) how to re-implement all this.

**Pull request closed without merging**